### PR TITLE
[r] Add support for registering arrays with `write_soma()`

### PR DESCRIPTION
**Pull request by mojaveazure**
This PR adds a `key` parameter to three `write_soma()` methods enabling automatic registration of an array into the parent collection. The following methods have been modified
- `write_soma.data.frame()`: for `data.frame` and derived objects (eg. `tibble`)
- `write_soma.matrix()`: for dense matrices
- `write_soma.TsparseMatrix()`: for sparse matrices

By default, `key` is set to `NULL`, which disables registration and preserves existing behavior

### CODE REVIEW

**Review by eddelbuettel — APPROVED** [review]
:shipit:

File: apis/r/DESCRIPTION [hunk 1]
```diff
@@ -6,7 +6,7 @@ Description: Interface for working with 'TileDB'-based Stack of Matrices,
     like those commonly used for single cell data analysis. It is documented at
     <https://github.com/single-cell-data>; a formal specification available is at
     <https://github.com/single-cell-data/SOMA/blob/main/abstract_specification.md>.
-Version: 1.6.99
+Version: 1.6.99.9001
```
Comment (eddelbuettel) [inline apis/r/DESCRIPTION:9]: How about 1.6.99.1 or 1.6.999 ?

Comment (johnkerl) [reply apis/r/DESCRIPTION]: @eddelbuettel I have no views on the matter. You said "pick a number, any number". I am happy to do whatever, in my clerical role; I have no prior R-opinions on the matter. @eddelbuettel @aaronwolen @mojaveazure as the R experts: please decide, and I'll type up whatever you decide in our [procedure](https://github.com/single-cell-data/TileDB-SOMA/wiki/Branches-and-releases).

Comment (mojaveazure) [reply apis/r/DESCRIPTION]: As discussed, this will be 1.6.99.1 to signify a development update (changed in 853ddd7b55890bad076926942e7eac96889b117e)

Comment (johnkerl) [reply apis/r/DESCRIPTION]: Thanks @mojaveazure !

https://github.com/single-cell-data/TileDB-SOMA/wiki/Branches-and-releases/_compare/5a2aa1fa0395ba84274d3ea191fe2b2641eb5c52...6d085ed4922577ce9acc71eac09df9181ae515f8

**Comment by johnkerl** [comment]
https://github.com/single-cell-data/TileDB-SOMA/pull/2018 manually

**Pull request merged**